biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java

```diff
@@ -315,6 +321,27 @@ private void doRequireCapability(Annotation a) throws Exception {
 		Attrs attrs = getAttributes(a, "ns");
 		directivesAndVersion(attrs, "filter", "effective", "resolution");
 		
+		for (Entry<String,String> entry : attrs.entrySet()) {
+			boolean modified = false;
+			StringBuffer sb = new StringBuffer();
+
+			Matcher matcher = SIMPLE_PARAM_PATTERN.matcher(entry.getValue());
+			while (matcher.find()) {
+				modified = true;
+				String key = matcher.group(1);
+				String substitution = attrs.get(key);
+
+				if (SIMPLE_PARAM_PATTERN.matcher(substitution).find())
+					throw new IllegalArgumentException("nested substitutions not permitted");
+
+				matcher.appendReplacement(sb, substitution);
+			}
+			matcher.appendTail(sb);
```

Comment:
You could move the `appendTail` call inside the `if (modified)` test since there is no need to do it otherwise.